### PR TITLE
feat(dashboard): add azure ad provider settings

### LINKED ISSUE #1130: Add Azure AD provider to the dashboard

### PR DESCRIPTION
**Pull request by dipakparmar**
- feat(graphql): add azuread to signinmethods query
- feat(dashboard-settings): add azuread provider settings component
- feat(dashboard-settings): add azuread provider to settings

Closes #1130

### CODE REVIEW

**Comment by dipakparmar** [comment]
I couldn’t generate the graphql SDK using codegen as I don't have the access to required queries in the local instance. 

For the logo, if this seems good, i can add it https://worldvectorlogo.com/logo/azure-2

File: dashboard/src/components/settings/signInMethods/AzureADProviderSettings /AzureADProviderSettings.tsx [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+          icon="/logos/AzureAD.svg"
+          switchId="authAzureADEnabled"
+          showSwitch
+          enabled={authEnabled}
+          className={twMerge(
+            'grid-flow-rows grid grid-cols-2 grid-rows-2 gap-y-4 gap-x-3 px-4 py-2',
+            !authEnabled && 'hidden',
+          )}
+        >
+          {/* <BaseProviderSettings /> */}
```
Comment (szilarddoro) [inline dashboard/src/components/settings/signInMethods/AzureADProviderSettings /AzureADProviderSettings.tsx:110]:
```suggestion
```

It's only required for providers that are reusing the same basic form.

File: dashboard/src/components/settings/signInMethods/AzureADProviderSettings /AzureADProviderSettings.tsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+          fullWidth
+          hideEmptyHelperText
+          />
+          <Input
+          {...register('authAzureADClientSecret')}
+          name="authAzureADClientSecret"
+          id="authAzureADClientSecret"
+          label="Azure AD Client Secret"
+          placeholder="Azure AD Client Secret"
+          className="col-span-3"
```
Comment (szilarddoro) [inline dashboard/src/components/settings/signInMethods/AzureADProviderSettings /AzureADProviderSettings.tsx:127]:
```suggestion
          className="col-span-1"
```

**Review by szilarddoro — COMMENTED** [review]
Awesome! Thanks for your first contribution @dipakparmar  🙌🏼

The implementation looks good, I only have a few minor suggestions. Also, we need wait for the relevant parts to be added to the backend.

**Comment by szilarddoro** [comment]
> I couldn’t generate the graphql SDK using codegen as I don't have the access to required queries in the local instance.
> 
> For the logo, if this seems good, i can add it https://worldvectorlogo.com/logo/azure-2

@eugene-reuka could you please prepare an icon for us? 🙏🏼

**Comment by dipakparmar** [comment]
@szilarddoro I have corrected it. Thank you :)

File: dashboard/src/gql/sign-in-methods/getSignInMethods.gql [hunk 3]
```diff
@@ -85,5 +85,11 @@ query signInMethods($id: uuid!) {
     authWorkOsDefaultDomain
     authWorkOsDefaultOrganization
     authWorkOsDefaultConnection
+
+    # AzureAD
+    authAzureADEnabled
+    authAzureADClientId
+    authAzureADClientSecret
+    authAzureADTenantId
```
[the remark this answers is not in the document]
Comment (dipakparmar) [reply dashboard/src/gql/sign-in-methods/getSignInMethods.gql:93]: which backend, is it open-source or i can contribute to?

[the remark this answers is not in the document]
Comment (szilarddoro) [reply dashboard/src/gql/sign-in-methods/getSignInMethods.gql:93]: I don't know how, but I missed your comment, sorry. That backend is unfortunately not open-source, so this is something we'll need to generate once we have everything supported on our end. I don't have an ETA, but maybe @plmercereau has some info.

**Comment by guicurcio** [comment]
> Is it possible to configure Azure AD in nhost without this being merged? I see it's supported by [hasura-auth](https://github.com/nhost/hasura-auth) - just not yet exposed in the nhost admin UI - i think. Thanks!

If you are using the Nhost cloud, it's not currently possible to configure this -- this is because each app needs to be aware of what these variables mean

\`\`\`  
 authAzureADEnabled  
 authAzureADClientId  
 authAzureADClientSecret  
 authAzureADTenantId  
\`\`\`  
and this requires changes in the way we deploy these applications (infrastructure.) Good news is that every other requirement is mostly ready (hasura-auth, front-end, our db.) We'll keep this issue updated.

**Comment by guicurcio** [comment]
I'm thinking that for this we could add a boolean flag in our database indicating that that the backend is not ready (nhost cloud) this way we could merge it, and open another more specific issue for the required infra changes.

**Comment by szilarddoro** [comment]
Hey @dipakparmar! Sorry for not getting back to you regarding this feature earlier. We have changed how sign-in methods are managed on the Settings page, so this PR will most likely need to be reworked. As far as I know, the provider is now supported by Hasura Auth. Would you like to pick this up again? Let me know if you have any questions.

**Comment by dipakparmar** [comment]
Sure, @szilarddoro I'll have some time on the weekend to work on this! Thanks

File: examples/codegen-react-apollo/nhost/config.yaml [hunk 4]
```diff
@@ -13,7 +13,7 @@ services:
       postgres_password: postgres
       postgres_user: postgres
   auth:
-    image: nhost/hasura-auth:0.16.1
+    image: nhost/hasura-auth:0.16.2
```
Comment (szilarddoro) [inline examples/codegen-react-apollo/nhost/config.yaml:16]: Is this change related to the Azure AD introduction?

**Review by szilarddoro — COMMENTED** [review]
Hi! Sorry for intruding, but I updated the codegen stuff, adjusted the form, and added the logo. I think we can get rid of the documentation link for now. We can add that separately.

**Comment by szilarddoro** [comment]
As for the scope removal: At the moment, managing scopes is not supported anywhere on the UI. In the future, it will be supported by some kind of multi-select component, but at the moment we should just get rid of it, and let Hasura Auth decide the default scopes.

File: examples/codegen-react-apollo/nhost/config.yaml (hunk 4 above)
Comment (dipakparmar) [reply examples/codegen-react-apollo/nhost/config.yaml:16]: I think I got this when I merged the main in this branch! It seems unrelated and can be removed.

File: dashboard/src/components/settings/signInMethods/AzureADProviderSettings/AzureADProviderSettings.tsx [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+        <SettingsContainer
+          title="AzureAD"
+          description="Allows users to sign in with AzureAD."
+          slotProps={{
+            submitButton: {
+              disabled: !formState.isDirty || maintenanceActive,
+              loading: formState.isSubmitting,
+            },
+          }}
+          docsLink="https://docs.nhost.io/authentication/sign-in-with-azuread"
```
Comment (szilarddoro) [inline dashboard/src/components/settings/signInMethods/AzureADProviderSettings/AzureADProviderSettings.tsx:140]: Could you please remove `docsLink` and `docsTitle`?  We'll add it later.

Comment (dipakparmar) [reply dashboard/src/components/settings/signInMethods/AzureADProviderSettings/AzureADProviderSettings.tsx]: Done!

**Review by szilarddoro — APPROVED** [review]
Good stuff @dipakparmar! Thanks for your contribution. I'll merge your changes to an internal branch to run some tests.
